homeassistant/components/ecobee/climate.py

```diff
@@ -536,3 +605,54 @@ def hold_preference(self):
         # supported; note that this should not include 'indefinite'
         # as an indefinite away hold is interpreted as away_mode
         return "nextTransition"
+
+    def create_vacation(self, service_data):
+        """Create a vacation with user-specified parameters."""
+        vacation_name = service_data[ATTR_VACATION_NAME]
+        cool_temp = service_data[ATTR_COOL_TEMP]
+        heat_temp = service_data[ATTR_HEAT_TEMP]
+
+        start_date = service_data.get(ATTR_START_DATE)
+        start_time = service_data.get(ATTR_START_TIME)
+        end_date = service_data.get(ATTR_END_DATE)
+        end_time = service_data.get(ATTR_END_TIME)
+        fan_mode = service_data[ATTR_FAN_MODE]
+        fan_min_on_time = service_data[ATTR_FAN_MIN_ON_TIME]
+
+        if cool_temp < 35:
```

Comment:
Shouldn't we just check `hass.config.units.temperature_unit`?

Reply:
This is how it's done in the climate base component:
https://github.com/home-assistant/home-assistant/blob/770ad86f126eb890f6c4021dc72261af62acd862/homeassistant/components/climate/__init__.py#L523-L524

Reply:
We could check `temperature_unit` and then just assume that the user has specified the temp in that unit (a fairly safe assumption I think?:
```python
if self.hass.config.units.temperature_unit == TEMP_CELSIUS:
    cool_temp = convert_temperature(cool_temp, TEMP_CELSIUS, TEMP_FAHRENHEIT)
    heat_temp = convert_temperature(heat_temp, TEMP_CELSIUS, TEMP_FAHRENHEIT)
```

Reply:
Yeah. We don't need the check I think. Just call convert with the correct units.